forem/forem
# Tag moderation UI changes

## What type of PR is this? (check all applicable)


## Description

## Related Tickets & Documents



- Closes #19822 

## QA Instructions, Screenshots, Recordings

[Figma Designs](https://www.figma.com/file/yhg9HTZDFIeWuvBt9ygrss/Moderation?node-id=1846%3A5086&mode=dev)

[Demo Video](https://www.loom.com/share/d46bd34c009e4db5b71cdebf9bfdcf05)

### How to test?

1. Sign in as Admin
2. Create a new post (without any tags)
3. Visit this post and click on `Mod actions` icon (visible on left side below bookmark icons)
4. Inside the iFrame (pop-up menu) click on `Adjust Tags`
5. Now add/remove tags in this, basically play with it

1. Now again, sign in as a trusted member but without any tags assigned for moderation
2. In this you will notice there is no option `Adjust Tags` after following the same steps as above.

1. Now assign this user some tags to moderate, maybe just 1 or 2 tags only.
2. To add tags visit `http://localhost:3000/admin/member_manager/users/<user-id>` and check out `Tags` and its information how to add it.
3. Now again follow the same steps (2-5) mentioned at top.

### Output

**Admin User - Light mode & Dark mode**
| Before  | After |
| ------------- | ------------- |
| <img width="272" alt="Screenshot 2023-08-18 at 1 10 42 AM" src="https://github.com/forem/forem/assets/9396084/6df65929-ef39-4106-9652-c4623c4b83ee"> | <img width="277" alt="Screenshot 2023-08-18 at 1 14 34 AM" src="https://github.com/forem/forem/assets/9396084/9edc0345-6d4d-4ccf-bc7e-59dfb0e72095"> |
| <img width="254" alt="Screenshot 2023-08-18 at 1 10 58 AM" src="https://github.com/forem/forem/assets/9396084/bf83e513-0649-477a-8270-8595db87afc7"> | <img width="265" alt="Screenshot 2023-08-18 at 1 14 55 AM" src="https://github.com/forem/forem/assets/9396084/290d131c-afd9-49cf-88a5-264d3811abc6"> |
| <img width="255" alt="Screenshot 2023-08-18 at 1 11 35 AM" src="https://github.com/forem/forem/assets/9396084/c6a647f1-e57e-497d-ade4-65aa5f563c4d"> | <img width="260" alt="Screenshot 2023-08-18 at 1 19 41 AM" src="https://github.com/forem/forem/assets/9396084/950faf1c-2302-420e-ae29-55b6f1041b3a"> |
| <img width="255" alt="Screenshot 2023-08-18 at 1 11 56 AM" src="https://github.com/forem/forem/assets/9396084/c03280fb-ef4b-4fec-ab51-a75647c34dd2"> | <img width="255" alt="Screenshot 2023-08-18 at 1 15 53 AM" src="https://github.com/forem/forem/assets/9396084/df6f6568-9631-41ab-8204-4a2a5f8f9b18"> |
| <img width="255" alt="Screenshot 2023-08-18 at 1 18 00 AM" src="https://github.com/forem/forem/assets/9396084/cd9ea091-0aa7-4170-b8ab-634b6119c614"> | <img width="260" alt="Screenshot 2023-08-18 at 1 17 41 AM" src="https://github.com/forem/forem/assets/9396084/f09b7233-dabd-43a5-8000-dc1589a7902e"> |

**Tag Mod User**
<img width="250" alt="Screenshot 2023-08-18 at 1 21 32 AM" src="https://github.com/forem/forem/assets/9396084/65b77372-79fe-4b96-b219-dbfc574d5029"> <img width="250" alt="Screenshot 2023-08-18 at 1 21 50 AM" src="https://github.com/forem/forem/assets/9396084/500e1684-9f25-4aa7-816f-9d6285b5907e"> <img width="250" alt="Screenshot 2023-08-18 at 1 21 59 AM" src="https://github.com/forem/forem/assets/9396084/e3dd1b40-1d97-4f92-83ba-6a8038fbc42e">


### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
<img width="251" alt="Screenshot 2023-08-18 at 2 16 48 AM" src="https://github.com/forem/forem/assets/9396084/18d71e19-42ae-4fde-bc99-9974e425d4fa">

- [x] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues? (There are issues on these pages but none of them are related to this PR)
- [x] Color contrast tested? (Yes, it passes for all accessible items)

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?
